daemon/checkpoint.go

```diff
@@ -38,6 +38,11 @@ func (daemon *Daemon) CheckpointCreate(name string, config types.CheckpointCreat
 		return fmt.Errorf("Invalid checkpoint ID (%s), only %s are allowed", config.CheckpointID, validCheckpointNameChars)
 	}
 
+	checkDir := filepath.Join(daemon.repository, container.ID, "checkpoints", config.CheckpointID)
+	if stat, err := os.Stat(checkDir); err == nil && stat.IsDir() {
+		return fmt.Errorf(" checkpoint with name %s already exists for container %s", config.CheckpointID, container.Name)
+	}
+
 	err = daemon.containerd.CreateCheckpoint(container.ID, config.CheckpointID, checkpointDir, config.Exit)
 	if err != nil {
```

Comment:
Ideally, we should probably catch the error here, and make a nicer message, however, I'm not sure what the correct way is to get the error-code, and if they are defined;

```
ERRO[0014] Handler for POST /v1.26/containers/test/checkpoints returned error: Cannot checkpoint container test: rpc error: code = 6 desc = mkdir /var/lib/docker/containers/86292a6c7d41387bf1a95a2981e

ERRO[0311] Handler for POST /v1.26/containers/test2/checkpoints returned error: Cannot checkpoint container test2: rpc error: code = 2 desc = exit status 1: "criu failed: type NOTIFY errno 0\nlog file.....
```